package.json

```diff
@@ -177,7 +178,7 @@
     "terriajs-tiff-imagery-provider": "2.13.3",
     "thredds-catalog-crawler": "0.0.7",
     "ts-essentials": "^5.0.0",
-    "typescript": "~5.2.0",
+    "typescript": "^5.7.2",
```

Comment:
@na9da Since you've been quiet for a few days I had a quick look at upgrading and got a crash on a `use-at-your-own-risk` when running 'yarn gulp lint`.

The error is not present when using ESLint 8.57.0, so I think the easiest way forward for upgrading typescript-eslint plugin is that someone looks at #7323 which should be merged before #7150. There is nothing intricate with #7323, it's just switching to the regular recommended configuration and disabling the new rules from that configuration that fails, and then adding a bunch of underscores to unused variables.

Reply:
Thanks. I checked upgrading eslint and typescript-eslint. The new versions introduces a bunch of new errors, here's a rough split:

- 34  @typescript-eslint/no-empty-object-type - We'll have to disable these rules (at least inside Mixins)
- 326 @typescript-eslint/no-require-imports - I'll try to tackle this in the webpack5 PR
- 57 @typescript-eslint/no-unused-expressions - Could possibly fix in another PR to be merged before this one

Overall it looks like we should first merge webpack5 and then this one.

Reply:
I rebased the ESLint 8 PR to latest main after you merged my unused variable fixes last week. I also see you have added the require-to-import conversion commit to the Webpack 5 PR, but I actually had something similar sitting around waiting for the CJS-to-ESM PR to get merged.

I do agree that it's probably better to get the ESLint upgrades merged even if that means some new rules have to be disabled; it's no worse than things are right now, and the disabled rules can be enabled one-by-one at a later stage.

As a general observation, I think you would have saved yourself some effort of having to fix merge conflicts if these other PRs with sweeping changes weren't hanging around in the queue for so long. The CJS-to-ESM PR is an obvious example of what I have in mind, and Zoran's react-imports change is another example. You might also get more help from others, I might have gotten around to upgrading typescript-eslint already if my ESLint 8 PR hadn't gotten stuck in the queue earlier.

Reply:
Now that ESLint 8 is merged, I put the update of typescript-eslint to the latest version in #7382.